src/storage/src/hummock/compaction_group_client.rs

```diff
@@ -132,6 +142,13 @@ impl MetaCompactionGroupClient {
         guard.set_index(compaction_groups);
         Ok(())
     }
+
+    async fn update_by(&self, compaction_groups: Vec<CompactionGroup>) -> HummockResult<()> {
```

Comment:
We can totally simplify MetaCompactionGroupClient, make it only updated by notification. 
And thus no need to refactor observer into async.

Reply:
> We can totally simplify MetaCompactionGroupClient, make it only updated by notification.
> And thus no need to refactor observer into async.

pull scheme will seldom work because we have push scheme and we only pull when UnknownTableId

Reply:
> We can totally simplify MetaCompactionGroupClient, make it only updated by notification.
> By removing all async lock from MetaCompactionGroupClient, we don't need to refactor observer into async.

fixed.